File: 01.1_req_rep/req.js
var zmq = require("zmq");

var request = zmq.socket("req");

request.connect("tcp://127.0.0.1:8011");
request.connect("tcp://127.0.0.1:8111");
request.connect("tcp://127.0.0.1:8211");

id = process.pid;
count = 0;

setInterval(function() {
  var t = request.send("msg nro. " + count++ + " (from " + id + ")");
  console.log(t);
}, 1000);

request.on("message", function(msg) {
  console.log("Response from: " + msg.toString());
});

File: 01.4_router/ruby/req.rb
require 'ffi-rzmq'

ctx = ZMQ::Context.new
request = ctx.socket ZMQ::REQ

request.connect("tcp://127.0.0.1:6000")
request.connect("tcp://127.0.0.1:6001")

count = 0

loop do
  request.send_string("msg nro.: #{count += 1}")
  # count += 1

  msg = ''
  request.recv_string(msg)
  puts "Response from: " + msg
  sleep 1
end
